src/NSL/logger.hpp

```diff
@@ -96,6 +103,50 @@ namespace NSL::Logger {
         }
     }
 
+    //! Initialize the logger
+    /*! 
+     * This function initializes the logger. 
+     * The parameter `log_level` must be either 
+     *  - "debug" (prints debug information)
+     *  - "info"  (prints regular run information) [DEFAULT]
+     *  - "warn"  (prints warnings and errors only)
+     *  - "error" (prints errors only)
+     *  The parameter `log_file` can be used to put the log output into
+     *  the specified file. If `log_file=""` output is logged to the console.
+     * */
+    inline void init(std::string log_level, std::string log_file){
+        std::map<std::string, spdlog::level::level_enum> levels = {
+            {"debug", spdlog::level::debug},
+            {"info", spdlog::level::info},
+            {"warn", spdlog::level::warn},
+            {"error", spdlog::level::err},
+        };
+
+        std::vector<spdlog::sink_ptr> sinks;
+        if(log_file != ""){
+            auto file_sink = std::make_shared<spdlog::sinks::basic_file_sink_mt>(log_file, true);
+            file_sink->set_level(levels["debug"]);
+            sinks.push_back(file_sink);
+        }
+
+        auto console_sink = std::make_shared<spdlog::sinks::stdout_color_sink_mt>();
+        console_sink->set_level(levels[log_level]);
+        sinks.push_back(console_sink);
+
+        auto logger = std::make_shared<spdlog::logger>("NSL_logger", begin(sinks), end(sinks));
+        logger->set_pattern("[%D %T] [%l] %v");
```

Comment:
Can we make this a default parameter, rather than hard-coded?  I'm happy with this format but don't see a reason to make a decision for all people for all time.

Reply:
Addressed in 617eea0ba7d05c19dfb4fe435978398f9a111659. Is that okay?